src/services/prices/price-sources/alchemy-price-source.ts

```diff
@@ -0,0 +1,126 @@
+import { ChainId, TimeString, Timestamp, TokenAddress } from '@types';
+import { IFetchService } from '@services/fetch/types';
+import { PriceResult, IPriceSource, PricesQueriesSupport, TokenPrice, PriceInput } from '../types';
+import { Chains, getChainByKeyOrFail } from '@chains';
+import { reduceTimeout, timeoutPromise } from '@shared/timeouts';
+import { filterRejectedResults, groupByChain, isSameAddress, splitInChunks } from '@shared/utils';
+import { Addresses } from '@shared/constants';
+
+const SUPPORTED_CHAINS: Record<ChainId, string> = {
```

Comment:
Just wondering if we could somehow reuse the list on the alchemy provider source. Not sure if it's the same or not to be honest